core/controllers/feedback_updates.py

```diff
@@ -0,0 +1,208 @@
+        else:
+            paginated_threads_list = self.normalized_payload[
+                'paginated_threads_list'
+            ]
+        if paginated_threads_list and paginated_threads_list[0]:
+            thread_summaries, number_of_unread_threads = (
+                feedback_services.get_exp_thread_summaries(
+                    self.user_id, paginated_threads_list[0]))
+        else:
+            thread_summaries, number_of_unread_threads = [], 0
+
+        self.values.update({
+            'thread_summaries': [s.to_dict() for s in thread_summaries],
+            'number_of_unread_threads': number_of_unread_threads,
+            'paginated_threads_list': paginated_threads_list[1:]
+        })
+        self.render_json(self.values)
+
+
+class FeedbackThreadHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """Gets all the messages in a thread."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'thread_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': constants.VALID_THREAD_ID_REGEX
+                }]
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {'GET': {}}
+
+    @acl_decorators.can_access_feedback_updates
+    def get(self, thread_id: str) -> None:
+        """Handles GET requests."""
+        assert self.user_id is not None
+        messages = feedback_services.get_messages(thread_id)
+        author_ids = [m.author_id for m in messages]
+        authors_settings = list(user_services.get_users_settings(author_ids))
+
+        message_ids = [m.message_id for m in messages]
+        feedback_services.update_messages_read_by_the_user(
+            self.user_id, thread_id, message_ids)
+
+        message_summary_list: List[
+            Union[MessageSummaryDict, SuggestionSummaryDict]
+        ] = []
+        suggestion = suggestion_services.get_suggestion_by_id(
+            thread_id, strict=False
+        )
+        suggestion_thread = feedback_services.get_thread(thread_id)
+
+        exploration_id = feedback_services.get_exp_id_from_thread_id(thread_id)
+        if suggestion:
+            suggestion_author_setting = user_services.get_user_settings(
+                author_ids[0], strict=True
+            )
+            if not isinstance(
+                suggestion,
+                suggestion_registry.SuggestionEditStateContent
+            ):
+                raise Exception(
+                    'No edit state content suggestion found for the given '
+                    'thread_id: %s' % thread_id
+                )
+            exploration = exp_fetchers.get_exploration_by_id(exploration_id)
+            current_content_html = (
+                exploration.states[
+                    suggestion.change.state_name].content.html)
```

Comment:
Is there a possibility of a KeyError in case "suggestion.change.state_name" does not exist in the exploration states (e.g. the exploration no longer has the state name referred to in the suggestion.)

Reply:
Same here Please see It is already written (https://github.com/oppia/oppia/blob/develop/core/controllers/learner_dashboard.py#L446)